drag_and_drop_v2/public/js/drag_and_drop_edit.js

```diff
@@ -183,12 +183,22 @@ function DragAndDropEditBlock(runtime, element) {
                         remove: function(e) {
                             var $el = $(e.currentTarget).closest('.zone-row'),
                                 classes = $el.attr('class'),
-                                id = classes.slice(classes.indexOf('zone-row') + 9);
+                                id = classes.slice(classes.indexOf('zone-row') + 9),
+                                // The id is of the form "zone-<index>".  We extract the index by
+                                // slicing off the first five letters and "casting" to integer.
+                                index = +id.slice(5),
+                                i;
```

Comment:
This works but seems quite fragile (not your fault, just how the code already was). I'd prefer something like: In the `add` method, before appending the new DOM node (`zoneTemplate(zoneObj) `), you can use `.data('index', num)` to save the index to the node as a JavaScript integer. Then to read the index here, it's just `$el.data('index')` with no casting or string processing needed.

Eventually, this code should be cleaned up a bit - data is stored redundantly in the DOM and in JavaScript, and there are several ways of identifying each zone (`id`, `index`, `i`, ...). It's quite confusing to read.